### ISSUE TITLE
<menu> not recognised as a valid parent for list items

### ISSUE THREAD

**Issue by Touffy** [dequelabs/axe-core#1962]
```html
<menu>
  <li><button>Edit</button></li>
  <li><button>Delete</button></li>
</menu>
```

Expectation: No error.

Actual: aXe complains that "`<li>` elements must be contained in a `<ul>` or `<ol>` (listitem)".

Motivation: [The HTML spec specifies that `<menu>` elements must have zero or more `<li>` children](https://html.spec.whatwg.org/multipage/grouping-content.html#the-menu-element).

<pre><code>
axe-core version: 3.3.2 using jest-axe
</code></pre>

**Comment by WilcoFiers** [comment]
This makes sense, but unfortunately the `<menu>` element isn't supported all that well, so using it can cause accessibility issues. I think the best way to do this is to use `role` attributes, both for the menu element and on the li elements. Axe-core supports this, and it should work well across various browsers / screen readers.

**Comment by Touffy** [comment]
If the `<menu>` element itself is an issue, then that's what aXe should report. Ignoring this issue just makes it look like aXe doesn't know HTML. Indeed, if I place the `<button>` elements as direct children of `<menu>` — which is incorrect HTML — I get no aXe error at all !

**Comment by runarberg** [comment]
See here: https://w3c.github.io/html-aria/#el-menu

> **Note:** some user agents suppress a list's [implicit ARIA semantics](https://w3c.github.io/html-aria/#dfn-implicit-aria-semantics) if list markers are removed. Authors can use `role=list` to reinstate the role if necessary.

So I guess to fix the accessibility concern authors should write:

```html
<menu role="list">
  <!-- `role="listitem"` is still implicit on the `<li>`s -->
  <li><button>Action 1</button></li>
  <li><button>Action 2</button></li>
  <li><button>Action 3</button></li>
</menu>
```

**Comment by runarberg** [comment]
In any case I find the current error message highly confusing and it doesn’t help me at all fixing the accessibility concern. On a different day I would have simply ignored this as a false positive (which it is; along with a false negative on the `<menu>` for failing to warn me about the lack of support)
